android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java

```diff
@@ -0,0 +1,128 @@
+ 
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.rate;
+
+import android.content.Context;
+import android.os.Build;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.OutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.chromium.base.Log;
+import org.chromium.base.ThreadUtils;
+import org.chromium.base.ContextUtils;
+import org.chromium.base.task.AsyncTask;
+import org.chromium.chrome.browser.settings.about.AboutChromeSettings;
+import org.chromium.chrome.browser.settings.about.AboutSettingsBridge;
+import org.chromium.chrome.browser.profiles.Profile;
+import org.chromium.chrome.browser.ntp_background_images.NTPBackgroundImagesBridge;
+
+public class RateFeedbackUtils {
+	private static final String TAG = "Rate";
+
+	public interface RateFeedbackCallback {
+		void rateFeedbackSubmitted();
+	}
+
+	public static class RateFeedbackWorkerTask extends AsyncTask<Void> {
+		private String mUserSelection;
+	    private String mUserFeedback;
+	    private RateFeedbackCallback mCallback;
+
+	    public RateFeedbackWorkerTask(String userSelection, String userFeedback, RateFeedbackCallback callback) {
+	    	mUserSelection = userSelection;
+	    	mUserFeedback = userFeedback;
+	        mCallback = callback;
+	    }
+
+	    @Override
+	    protected Void doInBackground() {
+	        sendRateFeedback(mUserSelection, mUserFeedback);
+	        return null;
+	    }
+
+	    @Override
+	    protected void onPostExecute(Void result) {
+	        assert ThreadUtils.runningOnUiThread();
+
+	        if (isCancelled()) return;
+
+	        mCallback.rateFeedbackSubmitted();
+	    }
+	}
+
+	private static void sendRateFeedback(String userSelection, String userFeedback) {
+		Context context = ContextUtils.getApplicationContext();
+		String appVersion = AboutChromeSettings.getApplicationVersion(context, AboutSettingsBridge.getApplicationVersion());
+		StringBuilder sb = new StringBuilder(); 
+		String http = "https://laptop-updates.brave.com/1/feedback";  
```

Comment:
do the hardcode on top of the class as final static for easy modifications in the future.